### PR TITLE
feat: add enable-debug flag

### PR DESCRIPTION
**Pull request by YeahNotSewerSide**
# Description

As default `/debug` endpoint is disabled on the node, and can be turned on with a supplied flag `--enable-debug`

[issue](https://github.com/vechain/protocol-board-repo/issues/266)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [x] Ran node in solo mode with debug enabled and disabled and called debug endpoints

**Test Configuration**:

* Go Version: go1.22.1 linux/amd64
* Hardware: 
* Docker Version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Comment by darrenvechain** [comment]
@YeahNotSewerSide looks like we need a PR to update the start command in e2e tests.

File: cmd/thor/flags.go [hunk 1]
```diff
@@ -150,6 +150,10 @@ var (
 		Value: "localhost:2112",
 		Usage: "metrics service listening address",
 	}
+	enableDebug = cli.BoolFlag{
```
Comment (leszek-vechain) [inline cmd/thor/flags.go:153]: can we add this flag to readme + some info about it there please 😄 maybe we can add some tests around it too

**Comment by YeahNotSewerSide** [comment]
> @YeahNotSewerSide looks like we need a PR to update the start command in e2e tests.

https://github.com/vechain/thor-e2e-tests/pull/28 - added a flag to the e2e tests

File: cmd/thor/flags.go [hunk 2]
```diff
@@ -150,6 +150,10 @@ var (
...
+		Name:  "enable-debug",
```
Comment (otherview) [inline cmd/thor/flags.go:154]: Suggestions to fall in line with the other flags:
- `enable-api-debug`
- `api-enable-debug`

File: cmd/thor/main.go [hunk 3]
```diff
@@ -95,6 +95,7 @@ func main() {
 			disablePrunerFlag,
 			enableMetricsFlag,
 			metricsAddrFlag,
+			enableDebug,
```
Comment (otherview) [inline cmd/thor/main.go:98]: Sugestion:
```suggestion
			enableAPIDebug,
```

**Review by otherview — DISMISSED** [review]
lgtm, worth perhaps adding the `api` word in the flag/vars to make more explicit.

I wonder if we should probably merge this once we have the https://github.com/vechain/protocol-board-repo/issues/267 ready too ? Will ask offline 👍

**Review by otherview — APPROVED** [review]
lgtm- Worth checking why the tests are failing.

**Comment by YeahNotSewerSide** [comment]
> I think we need to modify the tests to have debug enabled too (Darren suggested that)

https://github.com/vechain/thor-e2e-tests/pull/28 the pull request for that was open

**Comment by darrenvechain** [comment]
Closing as we went with a different approach

**Pull request closed without merging**